Sources/TestingMacros/Support/AttributeDiscovery.swift

```diff
@@ -133,6 +133,19 @@ struct AttributeInfo {
       return Argument(label: argument.label, expression: expr)
     }
 
+    // Any arguments of the test declaration macro which specify test arguments
+    // need to be wrapped a closure so they may be evaluated lazily by the
+    // testing library at runtime. If any such arguments are present, they will
+    // begin with a labeled argument named `arguments:` and include all
+    // subsequent unlabeled arguments.
+    if let argumentsIndex = otherArguments.firstIndex(where: { $0.label?.tokenKind == .identifier("arguments") }) {
+      for index in argumentsIndex ..< otherArguments.endIndex {
+        var argument = otherArguments[index]
+        argument.expression = .init(ClosureExprSyntax { argument.expression })
```

Comment:
This rewrites the arguments as detached nodes and will cause diagnostics we emit for them to fail to emit in the right location. Rewriting needs to be deferred to `functionArgumentList()`.

Reply:
Done!